src/ranch_acceptors_sup.erl

```diff
@@ -50,16 +50,22 @@ init([Ref, NumAcceptors, Transport]) ->
 	-> [{pos_integer(), inet:socket()}].
 start_listen_sockets(Ref, NumListenSockets, Transport, SocketOpts0, Logger) when NumListenSockets > 0 ->
 	BaseSocket = start_listen_socket(Ref, Transport, SocketOpts0, Logger),
-	{ok, Addr={_, Port}} = Transport:sockname(BaseSocket),
-	SocketOpts = case lists:keyfind(port, 1, SocketOpts0) of
-		{port, Port} ->
-			SocketOpts0;
-		_ ->
-			[{port, Port}|lists:keydelete(port, 1, SocketOpts0)]
+	{ok, Addr} = Transport:sockname(BaseSocket),
+	ExtraSockets = case Addr of
+		{local, _} when NumListenSockets > 1 ->
+			listen_error(Ref, Transport, SocketOpts0, reuseport_local, Logger);
+		{local, _} ->
+			[];
+		{_, Port} ->
+			SocketOpts = case lists:keyfind(port, 1, SocketOpts0) of
```

Comment:
Not sure if you're ok with the nested case, I can move that into a separate function if you want.